src/main/java/org/candlepin/controller/util/PromotedContent.java

```diff
@@ -0,0 +1,188 @@
+            return this;
+        }
+
+        log.debug("Checking for promoted content in environment: {}", environment.getName());
+        for (EnvironmentContent envContent : environment.getEnvironmentContent()) {
+            log.debug("  promoted content: {}", envContent.getContent());
+
+            this.contents.putIfAbsent(envContent.getContent().getId(), envContent);
+        }
+
+        return this;
+    }
+
+    /**
+     * Collects promoted content from all given environments.
+     *
+     * Expects environments to be sorted by priority (from highest to lowest) in
+     * order to deduplicate the promoted content.
+     *
+     * @param environments environments from which to collect promoted content
+     * @return an instance of {@link PromotedContent}
+     */
+    public PromotedContent withAll(List<Environment> environments) {
+        if (environments == null || environments.isEmpty()) {
+            log.debug("No environments to check for promoted content.");
+            return this;
+        }
+
+        for (Environment environment : environments) {
+            this.with(environment);
+        }
+
+        return this;
+    }
+
+    /**
+     * Returns an id of environment to which is the given {@link ProductContent} promoted.
+     *
+     * @param pc a product content for which to find environment
+     * @return an environment id
+     */
+    public String environmentIdOf(ProductContent pc) {
+        if (pc == null || pc.getContent() == null) {
+            return null;
+        }
+        String contentId = pc.getContent().getId();
+        EnvironmentContent environmentContent = this.contents.get(contentId);
+        if (environmentContent == null) {
+            log.debug("Environment not found for content: {}", contentId);
+            return null;
+        }
+        return environmentContent.getEnvironment().getId();
+    }
+
+    /**
+     * Checks if the given {@link ProductContent} is promoted into any environment.
+     *
+     * @param pc a product content to check
+     * @return true if the product content is promoted into any environment
+     */
+    public boolean contains(ProductContent pc) {
+        if (pc == null || pc.getContent() == null) {
+            return false;
+        }
+        return this.contents.containsKey(pc.getContent().getId());
+    }
+
+    /**
+     * Checks if the given product content is enabled.
+     *
+     * @param pc a product content to check
+     * @return true if the given product content is enabled
+     */
+    public Boolean isEnabled(ProductContent pc) {
+        if (!contains(pc)) {
+            return false;
+        }
+        return this.contents.get(pc.getContent().getId()).getEnabled();
+    }
+
+    /**
+     * Constructs a content path from environment specific prefix and given content.
+     *
+     * @param pc content for which to create a content path
+     * @return full content path
+     */
+    public String getPath(ProductContent pc) {
```

Comment:
I think one of the things @Ceiu was getting at, is that this method (in contrast to other public methods on this class), will still be used when a consumer is not in any environment, and it should still work, so we should update the javadoc of this method to explain that the path might be created with a prefix from either a) owner+environment, b) owner only or c) no prefix (just the content path)

Reply:
Also, we should unit test scenarios b) and c), since only scenario a) is unit tested in PromotedContentTest.java